### PR TITLE
feat(kit): support empty value in `InputCount`

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

### CODE REVIEW

**Review by nsbarsukov — APPROVED** [review]

File: projects/kit/components/input-count/input-count.component.ts [hunk 1]
```diff
@@ -233,7 +247,7 @@ export class TuiInputCountComponent
     }
 
     protected getFallbackValue(): number {
-        return 0;
+        return NaN;
```
Comment (nsbarsukov) [inline projects/kit/components/input-count/input-count.component.ts:250]: [InputNumber](https://taiga-ui.dev/components/input-number/API) is `null` when empty.
Probably, `InputCount` should have the same behavior 🤔 

Screenshot 2022-12-27 at 11 41 03

File: projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                .matchImageSnapshot(`02-input-count-empty-${empty}`);
+
+            cy.get(`#demo-content`)
+                .findByAutomationId(`tui-primitive-textfield__native-input`)
+                .focus()
+                .type(`1234`)
+                .blur();
+
+            cy.get(`#demo-content`)
+                .wait(WAIT_BEFORE_SCREENSHOT)
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts:29]: It generates this screenshot
input-count:input-count cy ts:__diff_output__:02-input-count-empty-true diff

Comment (nsbarsukov) [reply projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts:29]: Screenshot 2022-12-27 at 11 46 33

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts (hunk 2 above)
Comment (splincode) [reply projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts:29]: `It generates this screenshot` expected

Comment (splincode) [reply projects/demo-integrations/cypress/tests/desktop/kit/input-count/input-count.cy.ts:29]: image
fixed

File: projects/kit/components/input-count/input-count.component.ts (hunk 1 above)
Comment (splincode) [reply projects/kit/components/input-count/input-count.component.ts:250]: fixed

File: projects/kit/components/input-count/input-count.component.ts [hunk 3]
```diff
@@ -183,6 +183,8 @@ export class TuiInputCountComponent
         const capped = this.capValue(this.nativeNumberValue);
 
         if (capped === null || Number.isNaN(capped)) {
+            this.updateValue(NaN);
```
Comment (waterplea) [inline projects/kit/components/input-count/input-count.component.ts:186]: null

Comment (splincode) [reply projects/kit/components/input-count/input-count.component.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**